iroh-net/src/net/udp.rs

```diff
@@ -0,0 +1,150 @@
+use std::net::SocketAddr;
+
+use anyhow::{ensure, Context, Result};
+use tracing::warn;
+
+use super::Network;
+
+/// Wrapper around a tokio UDP socket that handles the fact that
+/// on drop `libc::close` can block for UDP sockets.
+#[derive(Debug)]
+pub struct UdpSocket(Option<tokio::net::UdpSocket>);
+
+/// UDP socket read/write buffer size (7MB). The value of 7MB is chosen as it
+/// is the max supported by a default configuration of macOS. Some platforms will silently clamp the value.
+const SOCKET_BUFFER_SIZE: usize = 7 << 20;
+impl UdpSocket {
+    /// Bind only Ipv4 on any interface.
+    pub async fn bind_v4(port: u16) -> Result<Self> {
+        Self::bind(Network::Ipv4, port).await
+    }
+
+    /// Bind only Ipv6 on any interface.
+    pub async fn bind_v6(port: u16) -> Result<Self> {
+        Self::bind(Network::Ipv6, port).await
+    }
+
+    /// Bind only Ipv4 on localhost.
+    pub async fn bind_local_v4(port: u16) -> Result<Self> {
```

Comment:
This is binding to `*` just like the `bind_v4` one does?

Reply:
uups

Reply:
fixed